### ISSUE TITLE
NPE in PendingProcessMessageSubscriptionChecker

### ISSUE THREAD

**Issue by deepthidevaki** [camunda/zeebe#16609]
**Describe the bug**

```
"Uncaught exception in 'AsyncProcessingScheduleActor-3' in phase 'STARTED'. Continuing with next job."

java.lang.NullPointerException: Cannot invoke "io.camunda.zeebe.engine.state.message.ProcessMessageSubscription.isOpening()" because "subscription" is null
	at io.camunda.zeebe.engine.processing.message.PendingProcessMessageSubscriptionChecker.sendPendingCommand(PendingProcessMessageSubscriptionChecker.java:86) ~[zeebe-workflow-engine-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.engine.state.message.DbProcessMessageSubscriptionState.visitPending(DbProcessMessageSubscriptionState.java:175) ~[zeebe-workflow-engine-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.engine.processing.message.PendingProcessMessageSubscriptionChecker.checkPendingSubscriptions(PendingProcessMessageSubscriptionChecker.java:79) ~[zeebe-workflow-engine-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:96) ~[zeebe-scheduler-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47) [zeebe-scheduler-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:122) [zeebe-scheduler-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
"
```

**Impact**
Only happened once. The partition was healthy afterwards, and continued to process with no issues.
Not sure if `PendingProcessMessageSubscriptionChecker` continued to run. 

**Logs**

[link](https://console.cloud.google.com/logs/query;cursorTimestamp=2024-02-26T13:04:16.615906160Z;endTime=2024-02-26T13:34:46.615Z;pinnedLogId=2024-02-26T13:04:16.615906160Z%2Fttlan475z65chldw;query=logName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.location%3D%22europe-west1%22%0Aresource.labels.namespace_name%3D%22e854f9ed-af8b-4cc2-a94f-50b9c14de75e-zeebe%22%0Aresource.labels.cluster_name%3D%22worker-chaos-1%22%0Aresource.labels.container_name%3D%22zeebe%22%0Aresource.labels.project_id%3D%22camunda-saas-int-chaos%22%0Aresource.labels.pod_name%3D%22zeebe-2%22%0AjsonPayload.context.partitionId%3D%223%22%0AjsonPayload.context.%22actor-scheduler%22%3D%22Broker-2%22%0A-jsonPayload.context.loggerName%3D%22io.atomix.raft.roles.LeaderAppender%22%0Atimestamp%3D%222024-02-26T13:04:16.615906160Z%22%0AinsertId%3D%22ttlan475z65chldw%22;startTime=2024-02-26T12:34:46.615Z?project=camunda-saas-int-chaos) 

There were no relevant logs related to this around that time. So I didn't collect them. 

**Environment:**
- Zeebe Version: 8.5.0-SNAPSHOT-main-e5c82370

**Comment by megglos** [comment]
ZPA-Triage:
- Hypotheses: could be a race condition between the task/checker thread and the stream processor - both have isolated transaction context on rocksDb
    - in this case, would be present since 8.2 then
- Impact: the checker could stop running after => we need to at least make it handle this situation without crashing
=> we need to check this

**Comment by korthout** [comment]
[Encountered on 8.4.1 again](https://console.cloud.google.com/errors/detail/CMaw17WHqY-QGQ;service=zeebe;time=P7D?project=camunda-cloud-240911). I think the fix [should also be backported to 8.4](https://github.com/camunda/zeebe/pull/16873#issuecomment-2004356683)
